lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -434,6 +438,40 @@ APInt getNextInputVal(Inst *Var,
   llvm::report_fatal_error("Model does not contain the guess input variable");
 }
 
+Inst *findConst(souper::Inst *I, std::string ConstName,
+                std::set<const Inst *> &Visited) {
+  if (I->Name == ConstName) {
+    return I;
+  } else {
+    for (auto &&Op : I->Ops) {
+      if (Visited.find(Op) == Visited.end()) {
+        auto Ret = findConst(Op, ConstName, Visited);
+        if (Ret) {
+          return Ret;
+        }
+      }
+    }
+    Visited.insert(I);
+  }
+  return nullptr;
+}
+
+bool exceeds64Bits(const Inst *I, std::set<const Inst *> &Visited) {
+  if (I->Width > 64) {
+    return true;
+  } else {
+    for (auto &&Op : I->Ops) {
+      if (Visited.find(Op) == Visited.end()) {
+        if (exceeds64Bits(Op, Visited)) {
+          return true;
+        }
+      }
+    }
+    Visited.insert(I);
```

Comment:
seems inefficient to insert I even if it was already there, is that the behavior you intended here?

Reply:
or nevermind, that's not really what it's doing.... I guess this is ok